engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy_test.go

```diff
@@ -446,3 +452,37 @@ func TestGetJsonLogStringWithEOFAndNoValidJsonEnding(t *testing.T) {
 	require.False(t, isComplete)
 	require.Equal(t, logLine1, jsonLogStr)
 }
+
+func TestParseTimestampFromJsonLogLineReturnsTime(t *testing.T) {
+	timestampStr := "2023-09-06T00:35:15-04:00"
+	jsonLogLine := map[string]string{
+		"timestamp": timestampStr,
+	}
+
+	expectedTime, err := time.Parse(time.RFC3339, timestampStr)
+	require.NoError(t, err)
+
+	actualTime, err := parseTimestampFromJsonLogLine(jsonLogLine)
+
+	require.NoError(t, err)
+	require.Equal(t, expectedTime, *actualTime)
+}
+
+func TestParseTimestampFromJsonLogLineWithIncorrectlyFormattedTimeReturnsError(t *testing.T) {
+	timestampStr := "2023-09-06" // not UTC formatted timestamp str
```

Comment:
This is a good test. Can we add another one to capture if the timezone information is absent like this:
`2023-09-06T00:35:15`